### PR TITLE
RenderBlock and RenderContent drafts

### PR DESCRIPTION
**Pull request by steve8708**
I think we can absolutely move all of the components into JSX lite

remaining features needing support to get there

- context support (medium)
- onUnmount hook (easy)
- useBuilderData custom async hook (medium)
- [emotion support](https://github.com/egoist/vue-emotion) (medium)
- [dynamic component support](https://vuejs.org/v2/guide/components.html#Dynamic-Components) (easy)
- move logic like stringToFunction, getBlockOptions, applyPatchWithMinimalMutationChain, and getEmotionCss from the react sdk to our utils package (medium)

cc @mandx for our todo list ^ :)

### CODE REVIEW

File: packages/core/src/__tests__/data/blocks/builder-render-content.raw.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          }
+          if (!(data && data.data)) {
+            break;
+          }
+          const patches = data.data[content.data?.id];
+          if (!(patches && patches.length)) {
+            return;
+          }
+
+          if (location.href.includes('builder.debug=true')) {
```
Comment (mandx) [inline packages/core/src/__tests__/data/blocks/builder-render-content.raw.tsx:41]:
```suggestion
          if (location.href.includes('builder.debugger=true')) {
```
The we could leave `builder.debug=true` for `console.log` stuff

**Review by mandx — COMMENTED** [review]
Nice, let me first check out this branch and play around :)

File: packages/core/src/__tests__/data/blocks/builder-render-content.raw.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          data-builder-model-name={props.model}
+          data-builder-component={content!.name}
+          data-builder-content-id={content!.id}
+          data-builder-variation-id={content!.testVariationId || content!.id}
+        >
+          <Show when={state.css}>
+            <style innerHTML={state.css} />¸
+          </Show>
+          <Show when={content!.data!.blocks}>
+            <For each={content!.data!.blocks}>
```
[the remark this answers is not in the document]
Comment (steve8708) [reply packages/core/src/__tests__/data/blocks/builder-render-content.raw.tsx:84]: that's not the case in JSX Lite, the elements are not constructed inside `Show` like they would be in react. it works like vue/angular/solid/svelte and other static template languages which makes this usage safe

image

File: packages/core/src/__tests__/data/blocks/builder-render-content.raw.tsx (hunk 1 above)
Comment (steve8708) [reply packages/core/src/__tests__/data/blocks/builder-render-content.raw.tsx:41]: oops I actually wanna get rid of this

File: packages/core/src/__tests__/data/blocks/builder-render-block.raw.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  useContext,
+  For,
+} from '@jsx-lite/core';
+import {
+  stringToFunction,
+  getBlockOptions, // TODO: migrate logic here
+  getEmotionCss, // TODO: migrate logic here
+} from '@builder.io/utils';
+
+type RenderBlockProps = {
```
[the remark this answers is not in the document]
Comment (steve8708) [reply packages/core/src/__tests__/data/blocks/builder-render-block.raw.tsx:17]: personally am a fan of `type __ =` in part bc it is just more terse 

lot of pros and cons overe here https://stackoverflow.com/questions/37233735/typescript-interfaces-vs-types

if you think one is definitely preferable to the other maybe add to team mtg list to discuss and see what we want to standardize on?

**Comment by steve8708** [comment]
going to get this in so we can work on top of it

**Pull request merged**